server/handlers/meshery_pattern_handler.go

```diff
@@ -184,82 +182,44 @@ func (h *Handler) handlePatternPOST(
-				k8sres, err = kompose.Convert(sourceContent) // convert the docker compose file into kubernetes manifest
-				if err != nil {
-					err = ErrConvertingDockerComposeToDesign(err)
-					return err
-				}
-
-			} else if sourcetype == string(models.K8sManifest) {
-				k8sres = string(sourceContent)
-			}
-			pattern, err := pCore.NewPatternFileFromK8sManifest(k8sres, "", false, h.registryManager)
-			if err != nil {
-				err = ErrConvertingK8sManifestToDesign(err)
-				return err
-			}
-			bytPattern, _ := yaml.Marshal(pattern)
-			mesheryPattern.PatternFile = string(bytPattern)
-		} else if sourcetype == string(models.HelmChart) {
-			// Write sourceContent to a temporary file
-			tempFile, err := os.CreateTemp("", "helm-chart-*.tgz")
-			if err != nil {
-				return fmt.Errorf("failed to create temp file: %w", err)
-			}
-			defer os.Remove(tempFile.Name()) // Ensure cleanup of the temporary file
-
-			_, err = tempFile.Write(sourceContent)
+		sourceContent := mesheryPattern.SourceContent
+		if len(mesheryPattern.SourceContent) == 0 {
+			h.log.Info("Pattern file doesn't contain SourceContent, fetching from remote provider")
+			sourceContent, err := provider.GetDesignSourceContent(token, mesheryPattern.ID.String())
 			if err != nil {
-				return fmt.Errorf("failed to write to temp file: %w", err)
-			}
-
-			err = tempFile.Close()
-			if err != nil {
-				return fmt.Errorf("failed to close temp file: %w", err)
+				return ErrDesignSourceContent(err, "get ")
 			}
+			mesheryPattern.SourceContent = sourceContent
+		}
 
-			// Use the temporary file path as LocalPath
-			latestKuberVersion := getLatestKubeVersionFromRegistry(h.registryManager)
-			resp, err := kubernetes.ConvertHelmChartToK8sManifest(kubernetes.ApplyHelmChartConfig{
-				LocalPath:         tempFile.Name(),
-				KubernetesVersion: latestKuberVersion,
-			})
-			if err != nil {
-				return ErrConvertingHelmChartToDesign(err)
-			}
+		fileToImport := FileToImport{
+			Data:     sourceContent,
+			FileName: mesheryPattern.Name, // Use pattern name as filename, make sure extension is there
+		}
 
-			result := string(resp)
-			pattern, err := pCore.NewPatternFileFromK8sManifest(result, mesheryPattern.Name, false, h.registryManager)
-			if err != nil {
-				return ErrConvertingHelmChartToDesign(err)
-			}
-			bytPattern, _ := yaml.Marshal(pattern)
+		// This function requires a valid file extension in the filename to work.
+		// Note: Assuming MesheryPattern is already sanitized and identified, we are again going through
+		// sanitization and identification (which is a redundant step but it's a one off)
+		design, _, err := ConvertFileToDesign(fileToImport, h.registryManager, h.log)
+		if err != nil {
+			return fmt.Errorf("Failed to convert pattern: %w", err)
+		}
 
-			mesheryPattern.PatternFile = string(bytPattern)
+		bytPattern, err := encoding.Marshal(design)
+		if err != nil {
+			return fmt.Errorf("Failed to marshal pattern: %w", err)
```

Comment:
need to use meshkit errrors , ConvertFileToDesign should already return a properly formatted meshkit error

Reply:
Got it, encoding.Marhsal also returns meshkit error. Simply returning error from here would do.